plugin/inlay_hint.py

```diff
@@ -112,40 +110,54 @@ def format_inlay_hint_tooltip(tooltip: Optional[Union[str, MarkupContent]]) -> s
 
 
 def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    tooltip = format_inlay_hint_tooltip(inlay_hint.get("tooltip"))
     result = ""
     can_resolve_inlay_hint = session.has_capability('inlayHintProvider.resolveProvider')
     label = inlay_hint['label']
-    is_clickable = bool(inlay_hint.get('textEdits')) or can_resolve_inlay_hint
+    has_text_edits = bool(inlay_hint.get('textEdits'))
+    is_clickable = has_text_edits or can_resolve_inlay_hint
     if isinstance(label, str):
         if is_clickable:
-            inlay_hint_click_command = sublime.command_url('lsp_inlay_hint_click', {
-                'session_name': session.config.name,
-                'inlay_hint': cast(dict, inlay_hint),
-                'phantom_uuid': phantom_uuid
+            inlay_hint_click_command = sublime.command_url('lsp_on_double_click', {
+                'command': 'lsp_inlay_hint_click',
+                'args': {
+                    'session_name': session.config.name,
+                    'inlay_hint': cast(dict, inlay_hint),
+                    'phantom_uuid': phantom_uuid
+                }
             })
             result += '<a href="{command}">'.format(command=inlay_hint_click_command)
-        result += html.escape(label)
+        instruction_text = '\nDouble-click to insert' if has_text_edits else ""
+        result += '<span title="{tooltip}">{value}</span>'.format(
+            tooltip=tooltip + instruction_text,
```

Comment:
Won't we have an empty line at the beginning in case the `tooltip` is not provided?

Reply:
It seems like ST might be stripping leading newline internally but I have still added an explicit strip just in case.